src/types/date.rs

```diff
@@ -0,0 +1,119 @@
+use std;
+use std::fmt;
+use std::fmt::Debug;
+use std::error::Error;
+use neon_runtime;
+use neon_runtime::raw;
+use context::{Context};
+use context::internal::Env;
+use result::{JsResult, JsResultExt};
+use object::{Object};
+use handle::{Handle, Managed};
+use super::{Value, ValueInternal};
+
+/// A JavaScript Date object.
+#[repr(C)]
+#[derive(Clone, Copy, Debug)]
+pub struct JsDate(raw::Local);
+
+impl Value for JsDate { }
+
+impl Managed for JsDate {
+    fn to_raw(self) -> raw::Local { self.0 }
+
+    fn from_raw(_: Env, h: raw::Local) -> Self { JsDate(h) }
+}
+
+#[derive(PartialEq, PartialOrd, Clone, Debug)]
+pub struct DateError(DateErrorKind);
+
+impl DateError {
+    pub fn kind(&self) -> DateErrorKind {
+        self.0
+    }
+}
+
+impl fmt::Display for DateError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", self.0.as_str())
+    }
+}
+
+impl Error for DateError {}
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Debug)]
+pub enum DateErrorKind {
+    Overflow,
+    Underflow,
+}
+
+impl DateErrorKind {
+    fn as_str(&self) -> &'static str {
+        match *self {
+            DateErrorKind::Overflow => "Uncaught RangeError: Date overflow",
```

Comment:
These look like strange formatting. Why do they include `Uncaught RangeError`? At this point, they might be caught. It would also result in `Uncaught RangeError: UncaughtRangeError: Date Overflow` when thrown.

Can make this more simply `Date overflow`?